mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -45,47 +64,68 @@ func runIgnitionFailure(c cluster.TestCluster) {
 	}
 }
 
-func ignitionFailure(c cluster.TestCluster) error {
-	// We can't create files in / due to the immutable bit OSTree creates, so
-	// this is a convenient way to test Ignition failure.
-	failConfig, err := conf.EmptyIgnition().Render(conf.FailWarnings)
-	if err != nil {
-		return errors.Wrapf(err, "creating empty config")
+func runDualBootfsFailure(c cluster.TestCluster) {
+	if err := dualBootfsFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	failConfig.AddFile("/notwritable.txt", "Hello world", 0644)
+}
 
-	builder := platform.NewQemuBuilder()
-	defer builder.Close()
-	builder.SetConfig(failConfig)
-	err = builder.AddBootDisk(&platform.Disk{
-		BackingFile: kola.QEMUOptions.DiskImage,
-	})
+func runDualBootfsIgnitionFailure(c cluster.TestCluster) {
+	if err := dualBootfsIgnitionFailure(c); err != nil {
+		c.Fatal(err.Error())
+	}
+}
+
+// Read file and verify if it contains a pattern
+// 1. Read file, make sure it exists
+// 2. regex for pattern
+func fileContainsPattern(path string, searchPattern string) (bool, error) {
+	file, err := os.ReadFile(path)
 	if err != nil {
-		return err
+		return false, err
 	}
-	builder.MemoryMiB = 1024
-	builder.Firmware = kola.QEMUOptions.Firmware
+	// File has content, but the pattern is not present
+	match := regexp.MustCompile(searchPattern).Match(file)
+	if match {
+		// Pattern found
+		return true, nil
+	}
+	// Pattern not found
+	return false, nil
+}
+
+// Start the VM, take string and grep for it in the temporary console logs
+func verifyError(builder *platform.QemuBuilder, searchPattern string) error {
 	inst, err := builder.Exec()
 	if err != nil {
 		return err
 	}
 	defer inst.Destroy()
-
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+
 	defer cancel()
 
 	errchan := make(chan error)
 	go func() {
-		err := inst.WaitAll(ctx)
-		if err == nil {
-			err = fmt.Errorf("Ignition unexpectedly succeeded")
-		} else if err == platform.ErrInitramfsEmergency {
-			// The expected case
-			err = nil
+		resultingError := inst.WaitAll(ctx)
+		if resultingError == nil {
+			resultingError = fmt.Errorf("ignition unexpectedly succeeded")
```

Comment:
I guess this error should probably be more generic now, since it's testing more than just Ignition failures. E.g. maybe just

> initramfs unexpectedly succeeded

And the function name could be more specific, e.g. `verifyInitramfsFailure()`.

But not worth a respin.